### ISSUE TITLE
Make preventWDAAttachments true by default

### ISSUE THREAD

**Issue by ohadschn** [appium/appium#7933]
It is simply too dangerous to be left `false` by default. The vast majority of Appium users know nothing about this folder (`~/Library/Developer/Xcode/DerivedData/WebDriverAgent-xxx...xxx/Logs/Test` and specifically its `Attachments` subfolder), and are most likely taking the screenshots they care about explicitly in their test code anyway. One day however they will find out about it (when their disk space runs out), and it will be too late.

Here's what happened to me, after happily running appium tests on my CI system for weeks:
1. Yesterday, just when I'm about to start off my weekend, I find out an Appium machine has run out of space. It contains a 500GB HDD with basically nothing on it except OS X, Appium, and XCode, so I have no idea what's taking up all that space.
2. I fire up Disk Inventory X but it seems to get stuck before revealing the offending folder[s] (Having an HDD [rather than SSD] that slows the system to a crawl sure doesn't help).
3. So I tried like 5 other similar utilities, all seemed to get stuck. Luckily, one of them (JDiskReport) had the courtesy to report which folder it was scanning, namely the `...Logs/Test` folder mentioned above (funny side story - I also realized it was scanning an 8TB mounted network share I had mounted, so make sure you eject those before if you run into a similar situation).
4. I figured all those UUID type folders were the culprit and deleted them. Of course, I was wrong.
5. I noticed the `Attachments` folder but Finder showed it as empty. It was only after noticing Finder couldn't calculate its size that I realized it was simply unable to enumerate its contents. A quick web search confirmed my suspicions, as a few people were complaining about this folder's size.
6. Thankful for finally discovering the root cause of the issue, I happily fired the command line thinking I'm one `rm -rf` away from being home free. Emphasis on "thinking". Even something as trivial as `ls` didn't work, nor did any of the other commands suited for this situation I found on the web (`ls -1 -f` and `find`). They would all simply hang, seemingly forever. The only thing I was able to do was confirm my suspicion about the amount of files in the folder, by viewing the size of the directory entry (not the contents) using `ls -lh` (~220MB which apparently translates into millions of files).
7. After some more research it became clear to me that I was only left with two options: [1] run `rm -rf` and hope it finishes in some reasonable time (been waiting for 20 hours now, still working - people have reported running it for days before completing) [2] If that fails, re-image the machine (yes, it's that bad).

To wrap things up, I apologize for my rant, but I wanted to demonstrate just how dangerous this folder is and how crucial it is for `true` to be the default for the `preventWDAAttachments` capability. Cleaning up the UUID type log folders as well (e.g. keep last N days) would be nice bonus, but the `Attachments` folder is by far the  more serious issue. Thanks!

**Comment by mykola-mokhnach** [comment]
First of all one has to say "thanks" to Apple, which made ~~absolutely necessary~~ attachments and logs creation option non-configurable. 
Setting _preventWDAAttachments_ capability to _true_ will make XCTest to generate tons of "access denied" errors in device logs, which some users don't like (there are such reports on that space already), since this is just a "dirty" workaround. Also, there is _clearSystemFiles_ capability, which deletes the content of DeriverData folder on session cleanup stage. 

Anyway it's better to let users to decide. What if one uses these logs and screenshots to generate test reports and we just delete them silently?

**Comment by ohadschn** [comment]
I agree that it's better to let users decide, I'm just saying that the _default_ should be `true`. My reasoning is simple: 

1. If you are bothered with the "access denied" errors, then the moment you notice them you look them up here, find the capability and switch it off (BTW I think there should be a huge warnings in the docs about doing so).
2. If you're using the logs and screenshots for your reports, you will notice their absence immediately and again, look it up and quickly find the capability. BTW I would wager the vast majority of users don't do this, if only for the simple reason that appium wasn't even using XCUItest until not that long ago.
3. Finally, if you are the vanilla user who just runs his tests and takes the screenshots he wants (95% of appium users if I had to guess), you will inevidably run into the scenario I described above. You will simply not know what hit you and how to diagnose it. And even if and after you do (I consider myself a very experienced problem solver, and this was very painful), your only viable choice might be to re-image your machine.

So in short, I'm saying that `false` is both a better default for the vast majority of users, and a far less dangerous default (because it does no harm and you find out its side effects quickly and easily).

**Comment by mykola-mokhnach** [comment]
@imurchie What do you think?

**Comment by imurchie** [comment]
The permissions logging ought to be fixed in the latest version of the driver. 

I'm fine with the default being `true`.

**Comment by amedvedjev** [comment]
i met with issue when mine macMini 1Tb SSD became full :-). it is took 1,5 day command line operation to delete all logs.
now
```
rm -fR /Users/jenkinss/Library/Developer/Xcode/DerivedData/* || true
```

is mine first step :-) for sure

**Comment by ohadschn** [comment]
@amedvedjev only 1.5 days? what a lightweight... :) And you had an SSD so count your blessings!
My deletion process on the other machine is nearing 3 days and still going strong (took 26 hours on the first machine, where the folder was almost 3 times smaller).

BTW for anyone encountering this I recommend the following command to see some stats and progress: `rm -rvf /path/to/folder | pv -l > /dev/null`.

**Comment by mykola-mokhnach** [comment]
The change has been merged to 1.6.4-beta
